src/Context.Tests/ImmutableConventionWithRecordsTests.cs

```diff
@@ -0,0 +1,116 @@
+using System.Threading.Tasks;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Squadron;
+using Xunit;
+
+namespace MongoDB.Extensions.Context.Tests
+{
+    public class ImmutableConventionWithRecordsTests
+    {
+        public class SimpleRecordCase : IClassFixture<MongoResource>
+        {
+            private readonly MongoDbContextData _context;
+
+            public SimpleRecordCase(MongoResource mongoResource)
+            {
+                _context = CreateContext(mongoResource);
+            }
+
+            [Fact]
+            public async Task ApplyConvention_SerializeSuccessful()
+            {
```

Comment:
I know the triple A is in the InsertAndFind method, but maybe we mark it with a comment // Arrange & Act & Assert

Reply:
I added a comment on each usage of InsertAndFind